tests/test_05_actions.py

```diff
@@ -515,8 +516,40 @@ def test_revoke_key(self, pkspec):
         self.gpg_verify_key(key)
 
     @pytest.mark.run(after='test_revoke_key')
-    def test_revoke_key_with_revoker(self):
-        pytest.skip("not implemented yet")
+    @pytest.mark.parametrize('pkspec', pkeyspecs, ids=[str(a) for a, s in pkeyspecs])
+    def test_revoke_key_with_revoker(self, pkspec):
+        if pkspec not in self.keys:
+            pytest.skip('Keyspec {} not in keys; must not have generated'.format(pkspec))
+
+        # revoke the key
+        rev = self.keys[next(pks for pks in pkeyspecs if pks != pkspec)]
+        key = self.keys[pkspec]
+
+        with rev.unlock('This Password Has Been Changed') as ukey:
+            rsig = ukey.revoke(key, reason=RevocationReason.Retired, comment="But you're so oooold")
+
+        assert 'ReasonForRevocation' in rsig._signature.subpackets
```

Comment:
i guess the point of this is to verify that the key is revoked? if so, I would state it more clearly in a comment and mark that for replacement once `is_revoked()` (or similer public API) is implemented.